### PR TITLE
Add attendance to scastorage

### PR DESCRIPTION
**Pull request by FlyingDonkeys**
- will be incorporating attendance into the scastorage

### CODE REVIEW

**Review by Zeepheru — APPROVED** [review]
LGTM

**Comment by devivy** [comment]
Have merge conflicts

File: src/main/java/tahub/contacts/model/studentcourseassociation/StudentCourseAssociation.java [hunk 1]
```diff
@@ -13,31 +13,52 @@
...
+     * @param tutorial the tutorial associated with this association
+     * @param gradingSystem the GradingSystem used to manage this student's grades for the associated course
+     * @param attendance the Attendance instance associated with this association
+     */
+    public StudentCourseAssociation(Person student, Course course, Tutorial tutorial,
+                                    GradingSystem gradingSystem, Attendance attendance) {
+        this.student = student;
+        this.course = course;
+        this.tutorial = tutorial;
+        this.grades = gradingSystem;
```
Comment (FlyingDonkeys) [inline src/main/java/tahub/contacts/model/studentcourseassociation/StudentCourseAssociation.java:61]: Overloaded constructor to distinguish between creating new SCA and updating existing SCA.

File: src/test/java/tahub/contacts/model/studentcourseassociation/StudentCourseAssociationTest.java [hunk 2]
```diff
@@ -26,8 +26,72 @@
...
+    private static final Tutorial TEST_TUTORIAL_1 = new Tutorial(
+            "T16",
+            TEST_COURSE_1
+    );
+
+    private static final Tutorial TEST_TUTORIAL_2 = new Tutorial(
+            "T17",
+            TEST_COURSE_1
+    );
+
```
Comment (FlyingDonkeys) [inline src/test/java/tahub/contacts/model/studentcourseassociation/StudentCourseAssociationTest.java:59]: Components which can be reused when making testcases.

File: src/test/java/tahub/contacts/storage/JsonAdaptedStudentCourseAssociationTest.java [hunk 3]
```diff
@@ -147,7 +152,31 @@ void toModelType_nullCourseAndTutorial_throwsIllegalValueException() {
...
+        JsonAdaptedPerson invalidPerson = new JsonAdaptedPerson(INVALID_MATRICULATION_NUMBER, INVALID_NAME,
+                INVALID_PHONE, INVALID_EMAIL, INVALID_ADDRESS, VALID_TAGS);
+        JsonAdaptedCourse validCourse = new JsonAdaptedCourse(VALID_COURSE_CODE, VALID_COURSE_NAME);
+        JsonAdaptedTutorial validTutorial = new JsonAdaptedTutorial(VALID_TUTORIAL_ID, validCourse);
+
+        JsonAdaptedStudentCourseAssociation adapter = new JsonAdaptedStudentCourseAssociation(invalidPerson,
+                validCourse, validTutorial, VALID_ATTENDANCE);
+        assertThrows(IllegalValueException.class, adapter::toModelType);
+    }
+
```
Comment (FlyingDonkeys) [inline src/test/java/tahub/contacts/storage/JsonAdaptedStudentCourseAssociationTest.java:170]: New testcases.

**Pull request merged**